custom-options-plus.php

```diff
@@ -112,6 +113,77 @@ function cop_insert( $row ) {
 	);
 }
 
+
+function cop_get_last_available_id(){
+	global $wpdb, $COP_TABLE;
+	$dbname = $wpdb->dbname;
+	$result = $wpdb->get_results("
+		SELECT `AUTO_INCREMENT`
+		FROM  INFORMATION_SCHEMA.TABLES
+		WHERE TABLE_SCHEMA = '$dbname'
+		AND TABLE_NAME = '$COP_TABLE';
+	");
+	return $result[0]->AUTO_INCREMENT;
+}
+function cop_check_id_exists($item){
+	global $wpdb, $COP_TABLE;
+	$result = $wpdb->get_results("
+		SELECT id from $COP_TABLE WHERE id = $item->id AND label != '$item->label';
+	");
+	return count($result) ? true : false;
+}
+function cop_redefine_table_to_sequential(){
```

Comment:
I disagree here. Primary key should be always fixed.
As well, as an order can be done with a simple update for all items in order...so we don't need to run any reset, save in memory and so on...
A simple update() for them all. That would be pretty cheap.

PS: Add an extra field is not a big deal for a big table. update all fields, yep.